### ISSUE TITLE
Disable Window 7 Zoom From Applying To The Misc Cursor Image (1.9.0.0.9732)

### ISSUE THREAD

**Issue by Cernelius** [triplea-game/triplea#3265]
Revert Custom Cursor Scaling (1.9.0.0.9732)

What did you do to the custom cursors?

It used to scale to 48x48. Now I'm running 1.9.0.0.9732 and it looks like it goes to 36x36 or something.

If this is an intended change (down from 48x48 to 36x36 or whatever), not something random, please let's talk about it in a forum post (unless I overlooked), before deciding on such matters.

Or is this a bug on my end? Am I doing something wrong? I'm almost assuming it must be, as I can hardly believe it is something that anyone intended to do, as it doesn't make any sense, since you can get a cursor smaller than 48x48 by just having the actual image smaller, while keeping it at 48x48, leaving empty space.

???

**Comment by Cernelius** [comment]
This is a 48x48 cursor, I'm using in a map I'm making, that used to stay at those dimensions, and now it gets silly small, with this engine:
![cursor](https://user-images.githubusercontent.com/19562310/37257498-940ab1a4-256a-11e8-926e-67dfded6123e.gif)

**Comment by Cernelius** [comment]
Sorry, I cannot think of a map with 48x48 cursor in the repository.
But you can test it with "Conquest of the World":
https://github.com/triplea-maps/conquest_of_the_world
That one comes with a 48x48 cursor I made.
It used to display fine, but now it gets displayed smaller, other than badly zoomed.
If this is unintended, I hope it will be reverted.
If this is intended, I hope it will be discussed in forum.

**Comment by DanVanAtta** [comment]
Can you post a screenshot of the problem you see in `conquest` @Cernelius ? that would help make sure we're all seeing the same problem.

**Comment by Cernelius** [comment]
Can you tell me, on Windows 7, how do I take screenshots including the cursor? I click on "Print" or whatever that key is named in English, then paste in Gimp, but the cursor is never part of the screenshot.

**Comment by DanVanAtta** [comment]
I think this was the app i liked when using windows: https://app.prntscr.com/en/index.html

Knowing specifically it is the cursor that is the problem is helpful : )

**Comment by Cernelius** [comment]
I've installed that program now, but I see no way to have the cursor as part of the screenshot. How do I do it, if it is possible, using it?

**Comment by Cernelius** [comment]
If this might make matters simple, please just remove any king of resizing of custom cursors images (it used to resize to 48x48). That would be the best way, so that the mapmakers can have the dimensions they want.

**Comment by RoiEXLab** [comment]
@Cernelius could you check if the map scaling affects the size of the cursor?

**Comment by Cernelius** [comment]
The "Map Zoom", nor the "Unit Size", menu settings have any effect on the cursor, that appears always scaled down to 32x32.
It used to be always rescaled to 48x48, and I've no doubt that was better than what I see now, but, if feasible, having no rescaling at all (the custom cursor being just the dimension of its own image) would be the best.
You can test it with 1.8.0.9, but I'm thinking this should be something that happened very recently, or I would have almost surely already noticed it.

**Comment by Cernelius** [comment]
Nevermind, kind of, damnit. Now I understand that you never saw what I used to see.

I should have checked 1.8.0.9. It is the same now, for me.

I found out that what changed is that I used to have Windows at 150% general settings (in Personalization/Screen, or whatever that is in English (trying to translate)).

So, I now realize that the damn cursor has always been at 32x32, and it becomes 48x48 if you are on Windows at general settings 150%.

This means that, for example, the custom cursor of "Conquest of the World" displays at its original 48x48 picture dimension with Windows at 150%, while it gets into a badly zoomed crappy 32x32 in pretty much any other cases.

This is bad.

Can this behaviour still be considered a bug? I don't think the custom cursor should go with the general settings of the engine, or should it? Well, anyway, the zoom of the cursor is so bad that can't be good.

Also, and I understand it only now, this behaviour causes the centring of cursors point set not at X and Y both 0, like the one of "The Pact of Steel", to be off, as Windows at 150% makes the image bigger, but doesn't move the centring, set in map.properties.

I mean, Windows at 150% doesn't zoom the map at 150%, luckily, so the cursor shouldn't be zoomed 150% either, shouldn't it?

So, basically, unless this get changed, that cursor I made for "Conquest of the World" is very bad looking, since it is 48x48 that gets resized to 32x32 by an awful zoom. As well as the other 48x48 cursor pictures I made are just bloody pointless ugly (except for other people being on Windows at 150%).

So, should I reopen this as a Feature Request in forum or would it be feasible to just remove the 32x32 resize (letting the custom cursor being of its image actual size) and avoiding Windows applying its zoom on it (otherwise, my 48x48 cursors would, then, become horribly pixellated 72x72 on Windows 150%)?

Why nobody told me that the cursor of Conquest of the World was awful? :-P

**Comment by Cernelius** [comment]
This is how I supposed you to see the cursor of Conquest of the World, when the image retains its original 48x48 dimensions, and also how you actually see it on Windows when having 150% general zoom settings:
![48x48_cursor_example01](https://user-images.githubusercontent.com/19562310/37260088-585bd938-258f-11e8-8d84-51b8111f590f.png)
Instead, I suppose you all see (and always saw) a very bad looking and badly pixellated smaller 32x32 version of it, right?
Well, anyways, it doesn't matter for that map, as that is a closed chapter, but it would be still nice having the possibility of using 48x48 cursors, as I (wrongly) assumed was the case.
For my medieval sword, otherwise, I guess I can cut it into a 32x32 "daga".

**Comment by Cernelius** [comment]
I meant that side problem, you can test on "The Pact of Steel" with Windows at 150% is that, there, you have a cursor with the point not in the upper-left of the image, but set at:

```
map.cursor.hotspot.x=15
map.cursor.hotspot.y=15
```

in map.properties

Windows  7 at 150% increases the cursor by 150%, but I don't believe changes the above settings, so the "target" image appears targeting incorrectly.

This would be the custom cursor of the "The Pact of Steel" map, that applies to both games in it:
![cursor_tpos](https://user-images.githubusercontent.com/19562310/37260230-f5a0e8b8-2590-11e8-9142-e32e8504a296.gif)
The settings are, of course, meant to set the hotspot at about the centre of the image, that would be X,Y = 15,15. I see that cursor is 32x32, and this confirms this was always the intended size, but it becomes 48x48 in Windows at 150%, but doesn't target correctly, because I guess the 15,15 hotspot doesn't get increased to 22.5,22.5.
If somebody on Windows 7, or another version having a similar behaviour can verify what I'm saying.

In my opinion, the custom cursor should not be resized, by Windows, or whatever system, settings, and it would be nice if it never gets resized by TripleA, either, so that you can have cursors bigger than 32x32.

**Comment by Cernelius** [comment]
I'll change the title of the topic...

**Comment by Cernelius** [comment]
So, basically, I think there is a main item:

**The general zoom (Windows 7) should not apply to the custom cursor image, especially since it gets zoomed very badly.**

Or, if it should, then there is the issue that the hotspot, defined in map.properties, is not scaled the same; so "target" like cursors are off, and users on Windows at 150% can't target correctly while playing the two games of the "the_pact_of_steel" map.

If doing the above, I also suggest to getting rid of the forced resize to 32x32, letting mapmakers have whatever custom cursor dimension they want (so, having 32x32 only if the "cursor" image is of those dimensions), but this would be a feature request (I assumed the default was 48x48, since I was getting that with Windows at 150%, so I made cursors of that size, comprising the one of "Conquest of the World").

**Comment by DanVanAtta** [comment]
If I understand correctly, OS resize is not being taken into account? EG: the game think's it is using a 32x32 cursor image, but the image is actually displayed as 48x48

**Comment by Cernelius** [comment]
Yes. But, since the system zoom doesn't zoom anything else but the cursor (luckyly you don't see the map at 150%, if you have Windows at 150%), I think the cursor shouldn't be zoomed, either; adding the fact that the zoom is so bad that zooming render the cursor awful looking, in any case. The zoom for the cursor being so bad is another main reason why I think the original "cursor" image should always just be used at its original dimensions, anyways, instead of being forcefully resized at 32x32 (that basically means you can only have 32x32 cursor images, as you don't want to have that zoom).

On the other hand if, for whatever reasons, you want the cursor being +50% when using Windows at 150%, yes I believe this doesn't apply to the hotspot coordinate, setting target like cursors off.

**Comment by Cernelius** [comment]
This reminds me of that other, already solved, issue, when Windows at over 100% (a lot of Windows users have) made the production Window wrongly sized. @ron-murhammer solved that; I guess he remembers that case.

**Comment by Cernelius** [comment]
I think this makes the custom feature not really usable, as, if you make a cursor at 32x32, then, getting zoomed with Windows, it will just look awful and pixellated for all users having such settings, then better not having it.

Even withot the Windows zoom issue, only 32x32 is really awfully few pixels for getting toghether something serious (I was under the delusion it was 48x48).

I'm thinking I'm just dropping custom cursors from my maps until this feature is at the current state. The feature is just too poor anyways, also considering it only supports GIF images.

Still, if it would stay 32x32 for all users, no matter the zoom, something like this might be half decent:
![cursor_32](https://user-images.githubusercontent.com/19562310/37263361-14978e40-25a8-11e8-8501-d353ad622226.gif)
